src/parallel/StaticIrregDomainDecomposition.cpp

```diff
@@ -0,0 +1,120 @@
+/**
+ * @file StaticIrregDomainDecompostion.cpp
+ * @author amartyads
+ * @date 21.02.24
+ */
+
+#include "StaticIrregDomainDecomposition.h"
+#include "Domain.h"
+#include "utils/Logger.h"
+#include <fstream>
+#include <sstream>
+
+StaticIrregDomainDecomposition::StaticIrregDomainDecomposition(Domain* domain) : 
+	StaticIrregDomainDecomposition(domain, MPI_COMM_WORLD, {std::vector<unsigned int>{},{},{}}) {}
```

Comment:
not super happy with this one. Yes it works but usually default values should be specified at member declaration so any new constructors get the same by default.

This would mean you would only have to set domain in this Constructor and the constructor with more arguments calls the one with fewer.

Reply:
The issue here is that I'm not sure how to get it to work with MaMiCo.

Currently the non-mamico behaviour is:
1. Initialise the decomp by leveraging the default behaviour of `DomainDecomposition`, using empty weights, then fill in the weights with _gridSize, then update box dims
2. Read the XML, if weights given, goto 3, else 4
3. if weights given, read weights, update `_gridSize` from weights, update `_coords` from `_gridSize` by using `DomainDemposition::initMPIGridDims()`, then update box dims
4. if weights not given, keep decomp from step 1

The intended mamico behaviour is
1. Mamico reads its XML, and passes the weights to ls1
2. ls1 initialises the decomp by using actual grid sizes from the weights
3. the ls1 XML file is expected to not have a CSV, hence the decomp from mamico in step 1 is kept, hence weights are not updated

The ls1 behaviour refreshes the weights when the CSV is read, and does a reinit. But the Mamico version has to be a one-and-done. I can't really think of a way to do this with a reversed constructor call order, especially given how `DomainDecompisition.cpp` is structured.

Reply:
Ok it's no big deal and if it works why not...